CMakeLists.txt

```diff
@@ -187,6 +187,22 @@ OPTION(OPENSSL_SUPPORT "Include support for OpenSSL" 1)
 
 OPTION(SOCKET_API "Include the socket API" 0)
 
+OPTION(SWIG_PYTHON "Include SWIG Python bindings" 0)
+IF (SWIG_PYTHON)
+    FIND_PACKAGE(SWIG REQUIRED)
+    INCLUDE(${SWIG_USE_FILE})
+    FIND_PACKAGE(PythonLibs)
+    INCLUDE_DIRECTORIES(${PYTHON_INCLUDE_PATH})
+
+    INCLUDE_DIRECTORIES(${CMAKE_CURRENT_SOURCE_DIR})
+
+    SET(CMAKE_SWIG_FLAGS "")
+    SET(CMAKE_C_FLAGS   "${CMAKE_C_FLAGS}  -Wno-error=shadow -Wno-error=pedantic")
```

Comment:
The SWIG auto-generated code causes these warnings, so it's needed for compilation. I'm not quite sure, currently, how to avoid the source of the warnings or whether its even possible to avoid them without changing SWIG itself. Clever ideas/suggestions on the issue are certainly appreciated.